app/components/hearings-list-for-milestones-list.js

```diff
@@ -0,0 +1,66 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+
+export default class ProjectMilestoneComponent extends Component {
+  // @argument
+  milestone;
+
+  milestoneParticipantReviewLookup = {
+    BP: 'Borough President Review',
+    BB: 'Borough Board Review',
+    CB: 'Community Board Review',
+  }
+
+  // An array of disposition models that match the current milestone
+  @computed('milestone', 'milestone.project.dispositions')
+  get currentMilestoneDispositions() {
+    const currentMilestoneDispositions = [];
+
+    const milestone = this.get('milestone');
+    // ALL dispositions associated with a milestone's project
+    const dispositions = milestone.get('project.dispositions');
+    const milestoneParticipantReviewLookup = this.get('milestoneParticipantReviewLookup');
+
+    // iterate through ALL of the current project's dispositions
+    // IF a single disposition's assignment's dcpLupteammemberrole matches the
+    // current milestone's displayName, push that disposition into the new
+    // currentMilestoneDispositions array
+    dispositions.forEach(function(disposition) {
+      const participantType = disposition.get('assignment.dcpLupteammemberrole');
+
+      // milestoneParticipantReviewLookup[participantType] = e.g. `Community Board Review`
+      if (milestone.displayName === milestoneParticipantReviewLookup[participantType]) {
+        currentMilestoneDispositions.push(disposition);
+      }
+    });
```

Comment:
Similar to deduped-hearings-list, but it looks like you want a `filter` method here, which returns a new array also:

```javascript
const currentMilestoneDispositions = dispositions.filter((disposition) => {
 // do your filtering logic here, with the if cases
  const participantType = disposition.get('assignment.dcpLupteammemberrole');

  return milestone.displayName === milestoneParticipantReviewLookup[participantType];
});
```

You could even just `return` the call to `dispositions.filter` :)

Reply:
will do!